### PR TITLE
Fix batch translation

### LINKED ISSUE #272: Add attention mask to global attn at training/test.
I think the attention mask over source words isn't used in current version of OpenNMT-py, ie `GlobalAttention.applyMask` is never called. It was there in the previous one.
This makes training incorrect and decoding with batch_size > 1 terrible.

### LINKED ISSUE #318: Translation quality worsen with big batch size
Hi, 

When translating long texts, you want to increase batch size. If you do, the sentences will be padded with trailing dots to the length of the longest sentence in the batch. The bigger the batch size, the more trailing dots there are.  This completely destroy the translation quality for many sentences :/ 

For example, with batch size 256, I get 9 BLEU. With batch size 32, I got 24 BLEU on the same dataset and with the same model.

### PR DESCRIPTION
**Pull request by guillaumekln**
These changes correctly mask illegal attention weights during translation. It could easily be extended for training when source batches with variable lengths are required but it is not applied for now.

Fixes #272.
Fixes #318.

### CODE REVIEW

**Comment by srush** [comment]
Thanks Guillaume, looks good.
